alchemy/src/prisma/postgres/api.ts

```diff
@@ -0,0 +1,298 @@
+import type {
+  PrismaConnectionListResponse,
+  PrismaDatabase,
+  PrismaDatabaseBackupsResponse,
+  PrismaDatabaseListResponse,
+  PrismaDatabaseConnection,
+  PrismaErrorResponse,
+  PrismaPostgresAuthProps,
+  PrismaPostgresRegion,
+  PrismaProject,
+  PrismaProjectListResponse,
+  PrismaWorkspace,
+  PrismaWorkspaceListResponse,
+} from "./types.ts";
+
+/**
+ * Error thrown when the Prisma Postgres Management API returns a non-success response
+ */
+export class PrismaPostgresApiError extends Error {
+  readonly status: number;
+  readonly method: string;
+  readonly url: string;
+  readonly code?: string;
+  readonly responseBody?: unknown;
+
+  constructor(props: {
+    status: number;
+    method: string;
+    url: string;
+    message: string;
+    code?: string;
+    responseBody?: unknown;
+  }) {
+    super(props.message);
+    this.status = props.status;
+    this.method = props.method;
+    this.url = props.url;
+    this.code = props.code;
+    this.responseBody = props.responseBody;
+  }
+}
+
+/**
+ * Minimal client for the Prisma Postgres Management API
+ */
+export class PrismaPostgresApi {
```

Comment:
We've been trying to move away from hand-coding the API clients.

Does Prisma have a generated TypeScript SDK for its control plane?

@john-royal has been standardizing our API generation with hey-api. @john-royal, maybe you can take a look at their API spec and see how well it generates.

Reply:
Would it be an option to use the hand-coded API client initially, and switch to a prisma provided SDK later?